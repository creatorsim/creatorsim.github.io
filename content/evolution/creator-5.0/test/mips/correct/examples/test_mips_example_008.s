
#
# Creator (https://creatorsim.github.io/creator/)
#

.data

  matrixA:  .word 0x34000000, 0x34000000, 0x34000000, 0x34000000
            .word 0x23450000, 0x00432210, 0x77512120, 0x14141414
            .word 0x00000214, 0x44551122, 0xAABFF012, 0x77D00000
            .word 0x0000FF23, 0x12345678, 0x87654321, 0x33441124

  matrixB:  .float 0.0, 0.0, 0.0, 0.0
            .float 0.0, 0.0, 0.0, 0.0
            .float 0.0, 0.0, 0.0, 0.0
            .float 0.0, 0.0, 0.0, 0.0

.text
main:
        la $t0, matrixA
        la $t1, matrixB
        li $t2, 4
        li $t3, 4
        move $t4, $zero
        move $t5, $zero
    
loop1:  beq $t2, $t4, end1
loop2:  beq $t3, $t5, end2
        l.s $f0  ($t0)
        s.s $f0  ($t1)
        addi $t0, $t0, 4
        addi $t1, $t1, 4
        addi $t5, $t5, 1
        b loop2
end2:   addi $t4, $t4, 1
        move $t5, $zero
        b loop1
end1:   li $v0, 10
        syscall
