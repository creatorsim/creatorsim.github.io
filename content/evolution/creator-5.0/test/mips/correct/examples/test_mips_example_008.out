
cr[PC]:0x5c; ir[1,at]:0x200040; ir[2,v0]:0xa; ir[8,t0]:0x200040; ir[9,t1]:0x200080; ir[10,t2]:0x4; ir[11,t3]:0x4; ir[12,t4]:0x4; sfpr[f0]:0x33441124; dfpr[FP0]:0x3344112400000000; memory[0x20007f]:0x24; memory[0x20007e]:0x11; memory[0x20007d]:0x44; memory[0x20007c]:0x33; memory[0x20007b]:0x21; memory[0x20007a]:0x43; memory[0x200079]:0x65; memory[0x200078]:0x87; memory[0x200077]:0x78; memory[0x200076]:0x56; memory[0x200075]:0x34; memory[0x200074]:0x12; memory[0x200073]:0x23; memory[0x200072]:0xff; memory[0x20006d]:0xd0; memory[0x20006c]:0x77; memory[0x20006b]:0x12; memory[0x20006a]:0xf0; memory[0x200069]:0xbf; memory[0x200068]:0xaa; memory[0x200067]:0x22; memory[0x200066]:0x11; memory[0x200065]:0x55; memory[0x200064]:0x44; memory[0x200063]:0x14; memory[0x200062]:0x02; memory[0x20005f]:0x14; memory[0x20005e]:0x14; memory[0x20005d]:0x14; memory[0x20005c]:0x14; memory[0x20005b]:0x20; memory[0x20005a]:0x21; memory[0x200059]:0x51; memory[0x200058]:0x77; memory[0x200057]:0x10; memory[0x200056]:0x22; memory[0x200055]:0x43; memory[0x200051]:0x45; memory[0x200050]:0x23; memory[0x20004c]:0x34; memory[0x200048]:0x34; memory[0x200044]:0x34; memory[0x200040]:0x34; keyboard[0x0]:''; display[0x0]:''; 

